resources/options.js

```diff
@@ -17,45 +26,26 @@ function changeListener() {
 // Saves options to chrome.storage.
 // A message will confirm this to the user.
 function saveOptions() {
-    chrome.storage.sync.set({
-        // General
-        loggingEnabled: $('#logging').prop('checked'),
-        // Privacy
-        trackTabs: $('#tabs').prop('checked'),
-        trackComments:  $('#comments').prop('checked'),
-        trackPeerComments: $('#peerComments').prop('checked'),
-        trackFocus: $('#focus').prop('checked'),
-        // Security
-        hashUsername: $('#username').prop('checked'),
-        hashRepo: $('#repo').prop('checked'),
-        hashFile: $('#file').prop('checked')
-        // Hints
-    }, function() {
-        Materialize.toast("Options saved!", 2000);
-    });
+    var saveObject = {};
+    // Loops over all options and adds the current value to the chrome storage.
+    for (var i = 0; i < options.length; i++) {
+        saveObject[options[i]] = optionsElement(i).prop('checked');
+    }
+    chrome.storage.sync.set(saveObject, function() {});
 }
 
 // Restores the states of the checkboxes, using the preferences stored in chrome.storage.
 function restoreOptionsState() {
-    chrome.storage.sync.get({
-        loggingEnabled: Boolean,
-        trackTabs: Boolean,
-        trackComments: Boolean,
-        trackPeerComments: Boolean,
-        trackFocus: Boolean,
-        hashUsername: Boolean,
-        hashRepo: Boolean,
-        hashFile: Boolean
-    }, function(items) {
-        // Saved values from the chrome.storage
-        $('#logging').prop('checked', items.loggingEnabled);
-        $('#tabs').prop('checked', items.trackTabs);
-        $('#comments').prop('checked', items.trackComments);
-        $('#peerComments').prop('checked', items.trackPeerComments);
-        $('#focus').prop('checked', items.trackFocus);
-        $('#username').prop('checked', items.hashUsername);
-        $('#repo').prop('checked', items.hashRepo);
-        $('#file').prop('checked', items.hashFile);
+    var defaultObject = {};
+    // Loops over all options and casts them to Booleans.
+    for (var i = 0; i < options.length; i++) {
+        defaultObject[options[i]] = Boolean;
+    }
```

Comment:
Why are the default values for this object the class "Boolean"? Do we even need the default values?